### PR TITLE
ext_proc: Update the proto comment

### PR DESCRIPTION
**Pull request by tyxia**
Update the stale info in the proto.

The design has already been updated to: Only sent body and trailer if the processing mode is set to ``SEND`` and
they are present in original request/response.

Reference : 
https://github.com/envoyproxy/envoy/issues/28620
[doc](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/ext_proc/v3/ext_proc.proto#envoy-v3-api-msg-extensions-filters-http-ext-proc-v3-externalprocessor:~:text=Request%20trailers%3A%20Delivered%20if%20they%20are%20present%20and%20if%20the%20trailer%20mode%20is%20set%20to%20SEND.)

Fixes: #32029
Thanks @sanjaypujare for finding and reporting this.

### CODE REVIEW

**Comment by tyxia** [comment]
cc @yanavlasov  @yanjunxiang-google 

Since they were involved in this design update.

**Comment by tyxia** [comment]
/assign @yanavlasov  @yanjunxiang-google 

Also, add them as reviewers

File: api/envoy/service/ext_proc/v3/external_processor.proto [hunk 1]
```diff
@@ -98,21 +98,15 @@ message ProcessingRequest {
     // The HTTP trailers for the request path. Unless ``async_mode`` is ``true``, the server
     // must send back a TrailerResponse message or close the stream.
     //
-    // This message is only sent if the trailers processing mode is set to ``SEND``.
-    // If there are no trailers on the original downstream request, then this message
-    // will only be sent (with empty trailers waiting to be populated) if the
-    // processing mode is set before the request headers are sent, such as
-    // in the filter configuration.
+    // This message is only sent if the trailers processing mode is set to ``SEND`` and
+    // original downstream request has trailer.
```
Comment (abeyad) [inline api/envoy/service/ext_proc/v3/external_processor.proto:102]: nit: s/original/the original

File: api/envoy/service/ext_proc/v3/external_processor.proto [hunk 2]
```diff
@@ -98,21 +98,15 @@ message ProcessingRequest {
...
     // The HTTP trailers for the response path. Unless ``async_mode`` is ``true``, the server
     // must send back a TrailerResponse message or close the stream.
     //
-    // This message is only sent if the trailers processing mode is set to ``SEND``.
-    // If there are no trailers on the original downstream request, then this message
-    // will only be sent (with empty trailers waiting to be populated) if the
-    // processing mode is set before the request headers are sent, such as
-    // in the filter configuration.
+    // This message is only sent if the trailers processing mode is set to ``SEND`` and
+    // original upstream response has trailer.
```
Comment (abeyad) [inline api/envoy/service/ext_proc/v3/external_processor.proto:109]: nit: s/original/the original

**Review by adisuissa — COMMENTED** [review]
Thanks for updating this!
Are there tests for this behavior?
Was the behavior always like that and the API comment was wrong or was it due to a code change?

/wait

**Comment by tyxia** [comment]
> Thanks for updating this! Are there tests for this behavior? Was the behavior always like that and the API comment was wrong or was it due to a code change?
> 
> /wait

Yea, the code and test have been updated, for example, in https://github.com/envoyproxy/envoy/pull/28592
So there are already tests in place.

File: api/envoy/service/ext_proc/v3/external_processor.proto [hunk 3]
```diff
@@ -98,21 +98,15 @@ message ProcessingRequest {
     // The HTTP trailers for the request path. Unless ``async_mode`` is ``true``, the server
     // must send back a TrailerResponse message or close the stream.
     //
-    // This message is only sent if the trailers processing mode is set to ``SEND``.
-    // If there are no trailers on the original downstream request, then this message
-    // will only be sent (with empty trailers waiting to be populated) if the
-    // processing mode is set before the request headers are sent, such as
-    // in the filter configuration.
+    // This message is only sent if the trailers processing mode is set to ``SEND`` and
```
Comment (yanjunxiang-google) [inline api/envoy/service/ext_proc/v3/external_processor.proto:101]: Looks good to me for these changes.

Please also change the above line 94 to be "response body" as mentioned in issue 32029.

**Review by abeyad — APPROVED** [review]
/lgtm api

File: api/envoy/service/ext_proc/v3/external_processor.proto (hunk 3 above)
Comment (tyxia) [reply api/envoy/service/ext_proc/v3/external_processor.proto:101]: Good catch.

**Comment by tyxia** [comment]
Sorry @abeyad , I need the stamp again. (Just updated another typo).

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!
/lgtm api

**Review by abeyad — APPROVED** [review]

**Comment by yanjunxiang-google** [comment]
LGTM

**Review by yanavlasov — APPROVED** [review]

**Pull request merged**